src/test/java/io/aiven/connect/elasticsearch/ElasticsearchSinkFailureTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.connect.elasticsearch;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.connect.elasticsearch.bulk.BulkResponse;
+import io.aiven.connect.elasticsearch.jest.JestElasticsearchClient;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+
+public class ElasticsearchSinkFailureTest {
+
+    @Test
+    public void testRetryIfRecoverable() throws IOException {
+        final ElasticsearchSinkTask elasticsearchSinkTask = new ElasticsearchSinkTask();
+
+        final JestElasticsearchClient failingClient = Mockito.mock(JestElasticsearchClient.class);
+        final AtomicInteger apiCallCounter = new AtomicInteger(0);
+        when(failingClient.executeBulk(any())).thenAnswer(i -> {
+            final int numAttempt = apiCallCounter.incrementAndGet();
+            if (numAttempt < 3) {
```

Comment:
add a variable to reuse:
```suggestion
            if (numAttempt < numRetriesBeforeSucceding) {
```